# A Reproducible Genetic Variant Calling Workflow Across Multiple Species    
### Overview
This repository provides a comprehensive workflow, including configuration files and processed outputs, for genetic variant discovery across diverse species. The pipeline implements a standardized, modular, and fully reproducible framework to transform raw Next-Generation Sequencing (NGS) reads into high-quality variant datasets (VCFs). This project aims to advance scientific research by ensuring computational reproducibility, enabling performance benchmarking, and supporting open data sharing.

### Repository Structure
-	**datasets/** : Data preparation scripts (reference genomes, sample data, and SNP databases).
-	**install_workflow/** : Environment setup scripts and the variant calling pipeline.
-	**script/** : Primary analysis scripts for the genetic variant calling.
-	**supp/** : Supplementary data and documentation.

### Scientific Motivation
This project establishes a transparent pipeline for the genetic variant discovery across diverse species using parameterized workflows and curated datasets. By making all code and results publicly available, we ensure the complete reproducibility of the analyses presented in our associated manuscript.

### Reproducibility
To facilitate ease of use and verification, this repository includes:
-	**Installation Guide** : Comprehensive instructions for setting up the required software environment.
-	**Workflow Scripts** : Fully documented scripts with pre-defined parameters.
-	**Validation Datasets**: Example datasets provided for testing and pipeline verification.

### Contact
For inquiries regarding analytical methods, results, or technical support, please contact:
-	HyeonJung Lee (hyeon@kaist.ac.kr)
Korea Advanced Institute of Science and Technology (KAIST)
-	Dr. Sunhee Kim (king@kongju.ac.kr)
Kongju National University, South Korea
